realm/realm-library/src/main/cpp/io_realm_internal_OsCollectionChangeSet.cpp

```diff
@@ -140,11 +140,17 @@ JNIEXPORT jboolean Java_io_realm_internal_OsCollectionChangeSet_nativeIsRemoteDa
 JNIEXPORT jint JNICALL Java_io_realm_internal_OsCollectionChangeSet_nativeGetOldStatusCode(JNIEnv*, jobject, jlong native_ptr) {
     TR_ENTER_PTR(native_ptr)
     auto& change_set = *reinterpret_cast<CollectionChangeSetWrapper*>(native_ptr);
-    return (int8_t) change_set.get().partial_sync_old_state;
+    return static_cast<int8_t>(change_set.get().partial_sync_old_state);
 }
 
 JNIEXPORT jint JNICALL Java_io_realm_internal_OsCollectionChangeSet_nativeGetNewStatusCode(JNIEnv*, jobject, jlong native_ptr) {
     TR_ENTER_PTR(native_ptr)
     auto& change_set = *reinterpret_cast<CollectionChangeSetWrapper*>(native_ptr);
-    return (int8_t) change_set.get().partial_sync_new_state;
+    return static_cast<int8_t>(change_set.get().partial_sync_new_state);
```

Comment:
Expanding the size should be safe, but I can change it if you think it would make it more clear. I did this because the ObjectStore type is `int8_t` so it seemed more explicit to use it all the way through?